rmf_site_format/src/workcell.rs

```diff
@@ -289,6 +344,129 @@ impl Workcell {
         serde_json::ser::to_string_pretty(self)
     }
 
+    pub fn to_urdf(&self) -> Result<urdf_rs::Robot, WorkcellToUrdfError> {
+        let workcell = self;
+
+        let visuals_and_parents: Vec<(urdf_rs::Visual, _)> = workcell
+            .visuals
+            .iter()
+            .map(|v| {
+                let visual = &v.1.bundle;
+                let visual = urdf_rs::Visual {
+                    name: Some(visual.name.clone()),
+                    origin: visual.pose.into(),
+                    geometry: visual.geometry.clone().into(),
+                    material: None,
+                };
+                (visual, v.1.parent)
+            })
+            .collect();
+        let mut parent_to_visuals = HashMap::new();
+        visuals_and_parents.iter().for_each(|(visual, parent)| {
+            parent_to_visuals
+                .entry(*parent)
+                .or_insert_with(Vec::new)
+                .push(visual.clone());
+        });
```

Comment:
This (and the snippet below) is still a bit inefficient since you are iterating over a vector to create another vector, then you iterate over the new vector to create a hash map, also cloning each element twice in the process.

I think you can just skip the creation of the first vector altogether and compress the logic in a single iteration that creates the hashmap, something like:

```
        let mut parent_to_visuals = HashMap::new();
        for (parent, visual) in workcell.visuals.iter() {
            let visual = &visual.bundle;
            let visual = urdf_rs::Visual {
                name: Some(visual.name.clone()),
                origin: visual.pose.into(),
                geometry: visual.geometry.clone().into(),
                material: None,
            };
            parent_to_visuals
                .entry(*parent)
                .or_insert_with(Vec::new)
                .push(visual);
        }
```

This should remove the creation of the intermediate vector, as well as removing the additional `visual.clone()` to just move the value.

Reply:
Should be fixed in 6c5999c.